Build fail on Alpine linux

### Describe the bug

A description of what the bug is.

### Steps to reproduce the behavior
1. Building master on Alpine linux
2. View output
3. See error
```
localhost:~/shared/work/aeron/msquic/build# ninja
[1/27] Building C object src/platform/CMakeFiles/platform.dir/datapath_linux.c.o
FAILED: src/platform/CMakeFiles/platform.dir/datapath_linux.c.o
/usr/bin/cc  -I../src/inc -Iopenssl/include -DQUIC_PLATFORM_LINUX -fms-extensions -fPIC -pthread -Wl,--no-as-needed -ldl -DQUIC_EVENTS_LTTNG -DQUIC_LOGS_STUB -DQUIC_DISABLE_RESUMPTION -DQUIC_DISABLE_0RTT_TESTS   -Werror -Wall -Wextra -Wformat=2 -Wno-type-limits -Wno-unknown-pragmas -Wno-unused-value -MD -MT src/platform/CMakeFiles/platform.dir/datapath_linux.c.o -MF src/platform/CMakeFiles/platform.dir/datapath_linux.c.o.d -o src/platform/CMakeFiles/platform.dir/datapath_linux.c.o   -c ../src/platform/datapath_linux.c
../src/platform/datapath_linux.c: In function 'QuicDataPathWorkerThread':
../src/platform/datapath_linux.c:2187:13: error: implicit declaration of function 'TEMP_FAILURE_RETRY' [-Werror=implicit-function-declaration]
 2187 |             TEMP_FAILURE_RETRY(
      |             ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[2/27] Building C object src/platform/CMakeFiles/platform.dir/platform_linux.c.o
FAILED: src/platform/CMakeFiles/platform.dir/platform_linux.c.o
/usr/bin/cc  -I../src/inc -Iopenssl/include -DQUIC_PLATFORM_LINUX -fms-extensions -fPIC -pthread -Wl,--no-as-needed -ldl -DQUIC_EVENTS_LTTNG -DQUIC_LOGS_STUB -DQUIC_DISABLE_RESUMPTION -DQUIC_DISABLE_0RTT_TESTS   -Werror -Wall -Wextra -Wformat=2 -Wno-type-limits -Wno-unknown-pragmas -Wno-unused-value -MD -MT src/platform/CMakeFiles/platform.dir/platform_linux.c.o -MF src/platform/CMakeFiles/platform.dir/platform_linux.c.o.d -o src/platform/CMakeFiles/platform.dir/platform_linux.c.o   -c ../src/platform/platform_linux.c
../src/platform/platform_linux.c: In function 'QuicThreadCreate':
../src/platform/platform_linux.c:647:18: error: implicit declaration of function 'pthread_attr_setaffinity_np'; did you mean 'pthread_setaffinity_np'? [-Werror=implicit-function-declaration]
  647 |             if (!pthread_attr_setaffinity_np(&Attr, sizeof(CpuSet), &CpuSet)) {
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                  pthread_setaffinity_np
cc1: all warnings being treated as errors
[12/27] Building CXX object src/test/bin/CMakeFiles/msquictest.dir/quic_gtest.cpp.o
ninja: build stopped: subcommand failed.
```

### Expected vs actual behavior

A description of what you expected to happen and what actually happened.

--- a/src/platform/CMakeLists.txt
+++ b/src/platform/CMakeLists.txt
@@ -48,10 +48,12 @@ target_compile_options(platform PRIVATE
      $<$<CXX_COMPILER_ID:MSVC>:
         ${MSVC_WARNING_FLAGS}>)
 
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-    add_dependencies(platform MsQuicEtw)
-else()
-    target_link_libraries(platform ${LTTNGUST_LIBRARIES})
+if(QUIC_ENABLE_LOGGING)
+    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+        add_dependencies(platform MsQuicEtw)
+    else()
+        target_link_libraries(platform ${LTTNGUST_LIBRARIES})
+    endif()
 endif()
 
 if(QUIC_TLS STREQUAL "openssl")

--- a/src/platform/datapath_linux.c
+++ b/src/platform/datapath_linux.c
@@ -2172,14 +2172,14 @@ QuicDataPathBindingGetLocalMtu(
 }
 
 #ifndef TEMP_FAILURE_RETRY
-#define TEMP_FAILURE_RETRY(expression)                                  \
-        ({                                                              \
-            long int FailureRetryResult = 0;                            \
-            do {                                                        \
-                FailureRetryResult = (long int)(expression);            \
-            } while ((FailureRetryResult == -1L) && (errno == EINTR));  \
-            FailureRetryResult;                                         \
-        })
+#define TEMP_FAILURE_RETRY(expression)                              \
+    ({                                                              \
+        long int FailureRetryResult = 0;                            \
+        do {                                                        \
+            FailureRetryResult = (long int)(expression);            \
+        } while ((FailureRetryResult == -1L) && (errno == EINTR));  \
+        FailureRetryResult;                                         \
+    })
 #endif
 
 void*

--- a/src/platform/datapath_linux.c
+++ b/src/platform/datapath_linux.c
@@ -2176,8 +2176,8 @@ QuicDataPathBindingGetLocalMtu(
         ({                                                              \
             long int FailureRetryResult = 0;                            \
             do {                                                        \
-                FailureRetryResult = (long int)(expression)             \
-            } while ((FalureRetryResult == -1L) && (errno == EINTR));   \
+                FailureRetryResult = (long int)(expression);            \
+            } while ((FailureRetryResult == -1L) && (errno == EINTR));  \
             FailureRetryResult;                                         \
         })
 #endif

--- a/src/platform/datapath_linux.c
+++ b/src/platform/datapath_linux.c
@@ -2171,6 +2171,17 @@ QuicDataPathBindingGetLocalMtu(
 #endif
 }
 
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression)                                  \
+        ({                                                              \
+            long int FailureRetryResult = 0;                            \
+            do {                                                        \
+                FailureRetryResult = (long int)(expression)             \
+            } while ((FalureRetryResult == -1L) && (errno == EINTR));   \
+            FailureRetryResult;                                         \
+        })
+#endif
+
 void*
 QuicDataPathWorkerThread(
     _In_ void* Context

--- a/src/platform/platform_linux.c
+++ b/src/platform/platform_linux.c
@@ -637,7 +637,7 @@ QuicThreadCreate(
         return errno;
     }
 
-#ifdef __GLIBC__  // TODO figure out 
+#ifdef __GLIBC__ 
     if (Config->Flags & QUIC_THREAD_FLAG_SET_IDEAL_PROC) {
         QUIC_TEL_ASSERT(Config->IdealProcessor < 64);
         // There is no way to set an ideal processor in Linux, so just set affinity

--- a/src/platform/platform_linux.c
+++ b/src/platform/platform_linux.c
@@ -640,7 +640,7 @@ QuicThreadCreate(
 #ifdef __GLIBC__  // TODO figure out 
     if (Config->Flags & QUIC_THREAD_FLAG_SET_IDEAL_PROC) {
         QUIC_TEL_ASSERT(Config->IdealProcessor < 64);
-        // TODO - Set Linux equivalent of ideal processor.
+        // There is no way to set an ideal processor in Linux, so just set affinity
         if (Config->Flags & QUIC_THREAD_FLAG_SET_AFFINITIZE) {
             cpu_set_t CpuSet;
             CPU_ZERO(&CpuSet);
@@ -681,7 +681,7 @@ QuicThreadCreate(
 #ifndef __GLIBC__
     if (Status == QUIC_STATUS_SUCCESS && Config->Flags & QUIC_THREAD_FLAG_SET_IDEAL_PROC) {
         QUIC_TEL_ASSERT(Config->IdealProcessor < 64);
-        // TODO - Set Linux equivalent of ideal processor.
+        // There is no way to set an ideal processor in Linux, so just set affinity
         if (Config->Flags & QUIC_THREAD_FLAG_SET_AFFINITIZE) {
             cpu_set_t CpuSet;
             CPU_ZERO(&CpuSet);

--- a/src/platform/platform_linux.c
+++ b/src/platform/platform_linux.c
@@ -637,6 +637,7 @@ QuicThreadCreate(
         return errno;
     }
 
+#ifdef __GLIBC__  // TODO figure out 
     if (Config->Flags & QUIC_THREAD_FLAG_SET_IDEAL_PROC) {
         QUIC_TEL_ASSERT(Config->IdealProcessor < 64);
         // TODO - Set Linux equivalent of ideal processor.
@@ -654,6 +655,7 @@ QuicThreadCreate(
             // TODO - Set Linux equivalent of NUMA affinity.
         }
     }
+#endif
 
     if (Config->Flags & QUIC_THREAD_FLAG_HIGH_PRIORITY) {
         struct sched_param Params;
@@ -675,6 +677,26 @@ QuicThreadCreate(
             Status,
             "pthread_create failed");
     }
+
+#ifndef __GLIBC__
+    if (Status == QUIC_STATUS_SUCCESS && Config->Flags & QUIC_THREAD_FLAG_SET_IDEAL_PROC) {
+        QUIC_TEL_ASSERT(Config->IdealProcessor < 64);
+        // TODO - Set Linux equivalent of ideal processor.
+        if (Config->Flags & QUIC_THREAD_FLAG_SET_AFFINITIZE) {
+            cpu_set_t CpuSet;
+            CPU_ZERO(&CpuSet);
+            CPU_SET(Config->IdealProcessor, &CpuSet);
+            if (!pthread_setaffinity_np(*Thread, sizeof(CpuSet), &CpuSet)) {
+                QuicTraceEvent(
+                    LibraryError,
+                    "[ lib] ERROR, %s.",
+                    "pthread_setaffinity_np failed");
+            }
+        } else {
+            // TODO - Set Linux equivalent of NUMA affinity.
+        }
+    }
+#endif
 
     pthread_attr_destroy(&Attr);
 

